core/base/src/TSystem.cxx

```diff
@@ -3580,6 +3623,24 @@ int TSystem::CompileMacro(const char *filename, Option_t *opt,
 
    // ======= Load the library the script might depend on
    if (result) {
+      if (compileIfDifferent) {
+         // Add to the dictionary the symbol which will be checked
+         // to avoid to rebuild, unload and reload the library if requested by the user
+         // with the "h" option
+         std::ofstream dictfile(dict, std::ios::app);
```

Comment:
On windows, this means reopening a file that was just written, it may induced race condition with the file system locks (or maybe not).   An alternative is to produce yet another file and have it included in the dictionary source file by `rootcling` by using, in the `LinkDef.h` the `#pragma extra_include ... ` line.